catalog/task/collect-data/0.1/tests/test-collect-data-with-extra-data.yaml

```diff
@@ -0,0 +1,149 @@
+---
+apiVersion: tekton.dev/v1beta1
+kind: Pipeline
+metadata:
+  name: test-collect-data-with-extra-data
+spec:
+  description: |
+    Run the collect-data task and verify that extraData task result is accurate.
+    Releases and ReleasePlans don't currently support extraData, so it is only
+    added to the ReleasePlanAdmission.
+  workspaces:
+    - name: tests-workspace
+  tasks:
+    - name: setup
+      taskSpec:
+        steps:
+          - name: create-crs
+            image: quay.io/hacbs-release/release-utils:4d8649dbb2b626f5fe9f4ff83c1bc3be268fad31
+            script: |
+              #!/usr/bin/env sh
+              set -eux
+
+              cat > release << EOF
+              apiVersion: appstudio.redhat.com/v1alpha1
+              kind: Release
+              metadata:
+                name: release-sample
+                namespace: default
+              spec:
+                snapshot: foo
+                releasePlan: foo
+              EOF
+              kubectl apply -f release
+
+              cat > releaseplan << EOF
+              apiVersion: appstudio.redhat.com/v1alpha1
+              kind: ReleasePlan
+              metadata:
+                name: releaseplan-sample
+                namespace: default
+              spec:
+                application: foo
+                target: foo
+              EOF
+              kubectl apply -f releaseplan
+
+              cat > releaseplanadmission << EOF
```

Comment:
You could save the yaml in the workspace somewhere (a subdir maybe) and that way you could compare the whole `extraData` field directly rather then writing it again. Like this: https://mikefarah.gitbook.io/yq/usage/tips-and-tricks#comparing-yaml-files

But it's up to you.

Reply:
I see. But yeah, maybe it's not worth it - you already rewrote your `extraData` as json anyway and it's simple enough that this is probably fine for now.